### PR TITLE
Logs cli utilities

### PR DESCRIPTION
**Pull request by danielSanchezQ**

### CODE REVIEW

File: src/bin/cli/logs/compare.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+            load_logs_from_file(path)?
+        } else {
+            unreachable!()
+        };
+
+        let sentry_logs_data: Vec<SentryFragmentLog> = sentry_logs
+            .iter()
+            .enumerate()
+            .filter_map(
+                |(i, raw_log)| match raw_log.get("message").map(|v| v.as_str()).flatten() {
```
Comment (Zeegomo) [inline src/bin/cli/logs/compare.rs:71]: `flat_map` in place of `map(..).flatten()`?

File: src/bin/cli/logs/compare.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    },
+}
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct Compare {
+    #[structopt(long, conflicts_with = "sentry-logs")]
+    sentry_url: Option<Url>,
+
+    #[structopt(long, conflicts_with = "sentry-logs")]
```
Comment (Zeegomo) [inline src/bin/cli/logs/compare.rs:37]: Could this conflict be handled with an enum and a `structopt(flatten)`?

Comment (danielSanchezQ) [reply src/bin/cli/logs/compare.rs:37]: IIRC `conflicts_with` didn't work well with flatten attributes.  It happen before. But I'll double check.

File: src/bin/cli/logs/compare.rs (hunk 1 above)
Comment (danielSanchezQ) [reply src/bin/cli/logs/compare.rs:71]: It is an `Option`. I could flat_map earlier but I would have to compose a larger piece I think.

Comment (Zeegomo) [reply src/bin/cli/logs/compare.rs:71]: yeah sorry, I meant `and_then` (just a different name for a flatmap on `Option`/`Result`)

File: src/bin/cli/logs/mod.rs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    #[error(transparent)]
+    CompareError(#[from] compare::Error),
+}
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub enum Logs {
+    Sentry(sentry::SentryLogs),
+    Compare(compare::Compare),
```
Comment (mzabaluev) [inline src/bin/cli/logs/mod.rs:26]: Please add documentation / help strings to explain what the commands do.

Comment (danielSanchezQ) [reply src/bin/cli/logs/mod.rs:26]: Absolutely

File: src/bin/cli/logs/compare.rs (hunk 2 above)
Comment (danielSanchezQ) [reply src/bin/cli/logs/compare.rs]: I removed this option, it is better to use downloaded logs than directly use them, since it may be a bit slow.

**Comment by danielSanchezQ** [comment]
Supersede by #26

**Pull request closed without merging**